pkg/client/provider.go

```diff
@@ -25,8 +25,10 @@ import (
 	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
 )
 
+type syncClientFunc func(client client.Client) error
+
 type Provider struct {
-	Client         client.Client
+	k8sClient      client.Client
```

Comment:
Can we keep `Client` here as it is clear that a `client.Client` in our context is a k8s client.